src/runtime/entries/cloudflare-module.ts

```diff
@@ -69,6 +73,22 @@ export default {
       body,
     });
   },
+  // eslint-disable-next-line require-await
+  async queue(
+    batch: MessageBatch,
+    env: CFModuleEnv,
+    context: ExecutionContext
+  ) {
+    return nitroApp.hooks.callHook("cloudflare:queue", batch, env, context);
```

Comment:
To make sure forward/backward compatibility, would be nice to send them all as `{ batch, env, context }`

Reply:
Will try.
My main issue with this is that from what I can see, `return nitroApp.hooks.callHook("test", () => 1)` doesn't return 1 ?
Is there any other API suggestion here ?

Reply:
Right, since we are supposed to return something, should we do something like this ? 

```ts
    const hookResponse = { response: new Response() }
    await nitroApp.hooks.callHook("cloudflare-module:queue", {
      batch,
      env,
      context,
      hookResponse
    });
    return hookResponse.response
 ```

Reply:
Ah for handling, yes we might pass a callback like `sendResponse` (from the context object)

Reply:
```ts
export default defineNitroPlugin((app) => {
  app.hooks.hook("cloudflare:scheduled", (event) => {
    event.waitUntil(new Promise((resolve) => resolve("scheduled:cloudflare")));
  });

  app.hooks.hook(
    "cloudflare-module:scheduled",
    ({ controller, env, context }) => {
      context.waitUntil(
        new Promise((resolve) => resolve("scheduled:cloudflare-module"))
      );
    }
  );

  app.hooks.hook("cloudflare:queue", (event) => {
    event.sendResponse = () => "queued:cloudflare";
  });

  app.hooks.hook("cloudflare-module:queue", ({ batch, env, context }) => {
    context.sendResponse = () => "queued:cloudflare-module";
  });
});
```
This is what it looks like when used.